### ISSUE TITLE
[v2.4.5-p1] Stock: "Only X left" not showing on product page

### ISSUE THREAD

**Issue by FadedOut** [magento/magento2#36362]
### Preconditions and environment

- Magento 2.4.5-p1 (upgrade from 2.4.4)
- PHP 8.1

*Tried the following (making no difference)
- (CSS) Minify - On/Off
- (HTML) Minify - On/Off
- (JS) Minify - On/Off
- Merge Files - On/Off
- (JS) Bundling - Off

### Steps to reproduce

1. Have settings configured like the following:

Only X left Threshold: 500
Display Products Availability in Stock on Storefront: Yes

![magento-show-stock-left-admin-config](https://user-images.githubusercontent.com/12685113/197478800-885fc998-29d4-4232-883d-89c5df432dcf.jpg)

2. Go to the frontend, go to a product and select a variation that has less than 500 (or whatever number was entered for the threshold above) in stock

### Expected result

When clicking a variation that is under the "stock threshold" (Only X left Threshold), it should pop up the remaining left under the text "IN STOCK".

### Actual result

When clicking a variation that is under the "stock threshold" (Only X left Threshold), nothing shows ("Only X left" does not show).

It used to show on 2.4.4.

...
This isn't a "show-stopper" issue, but it would be nice to have working properly. For customers that want to order in bulk it would be helpful to them to be able to tell if there are only a few remaining or if there are a lot remaining.
...

### Additional information

I've tested on stock "Luma" theme so I know it can't be because of my theme.

I have no other problems on the site, everything else is working (all js, visuals, etc.).

Any other ways for me to debug/test?

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [X] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by FadedOut** [comment]
@magento give me 2.4.5-p1 instance

**Comment by FadedOut** [comment]
Well shucks. Seems it works properly on a stock 2.4.5-p1. 

Maybe it was because I updated from 2.4.4 to 2.4.5-p1? If that shouldn't be the case, does anyone have any tips on how I could debug it to find out why it won't show for me?

I see in the HTML (`availability only configurable-variation-qty`) on the 2.4.5-p1 instance here it starts like this:

![magento-only-x-in-stock-fresh-245p1-no-variation-selected](https://user-images.githubusercontent.com/12685113/197514354-48a4e330-e448-422e-844c-4d749d418dc3.jpg)

^^^This is all mine does^^^, the CSS shows `display: none` and if I force to display `display: block` it says "Only %1 left" even if I select a variation or not.

It should change to this: 

![magento-only-x-in-stock-fresh-245p1-variation-selected](https://user-images.githubusercontent.com/12685113/197514542-5bf69bd2-e009-443e-9d92-def1202dede7.jpg)

But mine won't. As described above my site just stays on `display: none` and a %1 instead of the proper amount. 

Any ideas why it could be doing this? As mentioned in OP, everything else on the site is working properly and there are no console errors. Any tips/tricks that I could do to debug this?

I'd like for my customers to be able to see how many are in stock without just choosing a QTY then clicking add to cart only for the page to reload then to find out there aren't that many left (and without even knowing how many really are left because it doesn't say). 

My customers and I much appreciate your time in advance.

**Comment by engcom-Dash** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Dash** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Dash** [comment]
@magento give me 2.4.5-p1-develop instance

**Comment by engcom-Dash** [comment]
@magento give me 2.4-develop instance

**Comment by engcom-Dash** [comment]
Hi @FadedOut ,

We are checking this issue in 2.4-develop instance....We are confirming the issue...We are getting expected results

Here i attached screenshots for your reference:
<img width="1512" alt="Screenshot 2022-10-25 at 6 50 48 PM" src="https://user-images.githubusercontent.com/60198592/197785821-da53d744-908d-4ce0-bb77-adb364d66a7a.png">


Steps to reproduce:
1.Install Magento instance
2.Setup the configuration
3.Go to front end
4.Click the product and select the variant
5.Only X left is not showing .

We are getting the expected results...When we clicking the variant of product there is no popup displaying instead that that particular "Only X left" hided...its not displaying.So we are confirming the issue.


Let us know if we missing anything.

Thanks!!!

**Comment by aliraza512** [comment]
@magento give me magento2.4.5 p1 instant

**Comment by paulhaidon** [comment]
@FadedOut 

We upgraded from 2.4.4 to 2.4.5-p1 and also found this to be an issue. Very frustrating. Sometimes you can do the following to make it work:
- Set the product back in stock, save
- Set the product disabled, save
- Set the product enabled, save
- Frontend loads as expected (with product being in stock)

But this doesn't always work and I am still struggling to find a solution to getting it to work. I tried a fresh install and it didn't seem to have the issue. But that doesn't mean it isn't a factor. 

Took me a good while to find this but report so hoping others find it too as not sure what to do here.

**Comment by FadedOut** [comment]
> @FadedOut
> 
> We upgraded from 2.4.4 to 2.4.5-p1 and also found this to be an issue. Very frustrating.

So you're also having the "Only X left" not showing when it should be?

But indeed very frustrating. Especially because 
A) the magento devs think it's broken on a clean version of 2.4.5-p1 when it's not - it appears to be maybe just my site (or maybe  not) or maybe a bug from upgrading from 2.4.4 - so IDK what they're going to "fix". 
And B) this is a pretty important aspect of selling bulk products - the customer would be happy to be informed of how many products are left before clicking add to cart and having the page reload only to tell them there aren't that many left (then to not even give a number how many are left - bonkers). Who sells just 1 of every item in their store? I can guarantee not every store...

So this probably won't even get fixed because they aren't even looking in the right place [if it is an upgrade issue (from 2.4.4 -> 2.4.5-p1]). I explained above that it works properly on a clean 2.4.5-p1 but maybe it's during the upgrade. Or maybe it _is_ just my site because no one else has reported the same issue...idk. But that is why I was asking for insight

> Though our situation might be slightly different in that **the stock is set to out of stock from purchases**, then when trying to add more stock to the simple product (part of a configurable product), the frontend does not reflect the backend save.
> 
> Sometimes you can do the following to make it work:
> 
> * Set the product back in stock, save
> * Set the product disabled, save
> * Set the product enabled, save
> * Frontend loads as expected (with product being in stock)
> 
> But this doesn't always work and I am still struggling to find a solution to getting it to work. I tried a fresh install and it didn't seem to have the issue. But that doesn't mean the bug doesn't exist. It might be that testing should include a 2.4.4 instance and then upgrading it to 2.4.5-p1, not something I have tried.
> 
> Took me a good while to find this but report so hoping others find it too as not sure what to do here.

So this I'm confused on. Are you having a separate issue? An issue with stock showing in general on the product page? If so, these are two separate issues and might warrant it's own bug report. From what I can tell, the products are showing correctly on my store if they are in stock or out of stock. I also use a CSV import file to update my products, I have waay too many to manually update them 1-by-1. And [referring to the text I bolded above] do you mean allowing backorders? I don't allow backorders, so that is different between our stores and could be important to your issue here.

**Comment by paulhaidon** [comment]
We don't allow backorders, no. 

It may well be a separate issue, I thought it might be linked given the issue we also have relates to the stock issues which seem to have only arisen as a result of going from 2.4.4 to 2.4.5-p1. 

As you've mentioned it may not be the same however, I will create my own bug report and have that looked into.

**Comment by FadedOut** [comment]
Would anyone have any ideas as to what would make the "Only X left" not show up properly?

It works fine on a standard/stock Magento 2.4.5-p1 but I can't seem to figure out what could be doing it on my side. I tried the standard Luma theme and it still doesn't show. I don't have any modules that would mess with that area and I don't have any modules that mess with the Javascript (like deferring).

It worked on 2.4.4, but I jumped straight to 2.4.5-p1 and only checked this recently, but it seems to have maybe occurred after upgrading. Anyone else that went from 2.4.4 to 2.4.5-p1 is the "Only X left" working for you?

I greatly appreciate anyone's ideas, no matter how small - they could lead to the resolution. 

Thank you in advance!

**Comment by FadedOut** [comment]
Just like to update this for anyone else that may come across this issue. 

It seems it IS indeed related to a module that I had no idea would mess with this but it seems to do so. The module:

[Magaplaza Lazy Loading](https://github.com/mageplaza/magento-2-lazy-loading) 4.0.3

I made a bug report here: https://github.com/mageplaza/magento-2-lazy-loading/issues/20

It seems like maybe some type of javascript conflict maybe? I'm not completely sure. I had no clue that module could mess with that area of the product page but after much debugging I found that to be the culprit (complete removing the module solved it, simply disabling did not - very odd). 

So, I'm closing this ticket as I have no clue what the Magento devs are going to fix as I mentioned much earlier in this ticket I found out it had nothing to do with Magento, as it worked fine using the instance (2.4-dev as well as 2.4.5-p1).
